sympy/integrals/tests/test_integrals.py

```diff
@@ -402,6 +402,118 @@ def test_issue_13749():
 def test_issue_18133():
     assert integrate(exp(x)/(1 + x)**2, x) == NonElementaryIntegral(exp(x)/(x + 1)**2, x)
 
+def test_issue_21024():
+    x = Symbol('x', real=True, nonzero=True)
+
+    f = log(x)*log(4*x) + log(3*x + exp(2))
+    F1 = x*log(x)**2 + x*(1 - 2*log(2)) + (-2*x + 2*x*log(2))*log(x) + \
+        (x + exp(2)/6)*log(3*x + exp(2)) + exp(2)*log(3*x + exp(2))/6
+    F2 = integrate(f,x)
+    assert simplify(F1 - F2) == 0
+
+    f = (x + exp(3))/x**2
+    F1 = log(x) - exp(3)/x
+    F2 = integrate(f,x)
+    assert simplify(F1 - F2) == 0
+
+    f = (x**2 + exp(5))/x
+    F1 = x**2/2 + exp(5)*log(x)
+    F2 = integrate(f,x)
+    assert simplify(F1 - F2) == 0
```

Comment:
We shouldn't use `simplify` in tests as it can be slow.

Can this not just be `assert F1 == F2`?

If there is a need to `simplify` the difference then a specific simplification function should be used such as `expand` or `signsimp` or something.